base/broadcast.jl

```diff
@@ -440,4 +440,24 @@ for (sigA, sigB) in ((BitArray, BitArray),
     end
 end
 
+############################################################
+
+# x[...] .= f.(y...) ---> broadcast!(f, dotview(x, ...), y...).
+# The dotview function defaults to getindex, but we override it in
+# a few cases to get the expected in-place behavior without affecting
+# explicit calls to view.   (All of this can go away if slices
+# are changed to generate views by default.)
+
+dotview(args...) = getindex(args...)
+dotview(A::AbstractArray, args...) = view(A, args...)
+dotview{T<:AbstractArray}(A::AbstractArray{T}, args...) = getindex(A, args...)
```

Comment:
@stevengj Why did you decide to default to `getindex` and not `view` in this case? @alanedelman showed me this example today
```
s = zeros(2,2) 
A = fill(s, 4, 4)
A[1:3,1:3] .= [ones(2,2)]
```
which doesn't set `A` because of this.